collated/main/A0138601M.md

```diff
@@ -0,0 +1,966 @@
+    }
+    
+    private static String removePeriod(String time) {
+        return time.substring(TIME_COMPONENT_HOUR_START_INDEX, time.length() - TIME_COMPONENT_PERIOD_LENGTH);
+    }
+    
+    /**
+     * Parses string time input with continental (24hr) format into LocalTime time.
+     *
+     * @param string time input with 24hr format
+     * @return LocalTime time based on the string time input
+     */
+    private static LocalTime parseTimeWithContinentalFormat(String time) throws DateTimeException {
+        String[] timeComponents = time.split(":");
+
+        int hour, minute;
+        hour = Integer.parseInt(timeComponents[TIME_COMPONENT_INDEX_HOUR]);
+        minute = Integer.parseInt(timeComponents[TIME_COMPONENT_INDEX_MINUTE]); 
+        
+        return LocalTime.of(hour, minute);
+    }
+    
+}
+```
+###### /java/seedu/todolist/model/task/Interval.java
+``` java
+public class Interval implements Comparable<Interval> {
+    
+    public static final String MESSAGE_INTERVAL_CONSTRAINTS_DATE = "End date cannot be earlier than start date";
+    public static final String MESSAGE_INTERVAL_CONSTRAINTS_TIME = "End time cannot be earlier than start time";
+    
+    private TaskDate startDate;
+    private TaskTime startTime;
+    private TaskDate endDate;
+    private TaskTime endTime;
+
+    
+    public Interval() {
+        this.startDate = null; 
+        this.startTime = null;
+        this.endDate = null;
+        this.endTime = null;
+    }
+    /**
+     * Validates given interval.
+     *
+     * @throws IllegalValueException if given set of date is an invalid interval.
+     */
+    public Interval(String startDate, String startTime, String endDate, String endTime) throws IllegalValueException {
+        this();
+        if (!isFloat(startDate, startTime, endDate, endTime)) {
+            if (isDeadlineWithTime(startDate, startTime, endDate, endTime)) {
+                this.endDate = new TaskDate(endDate);
+                this.endTime = new TaskTime(endTime);
+            }
+            else if (isDeadlineWithoutTime(startDate, startTime, endDate, endTime)) {
+                this.endDate = new TaskDate(endDate);
+            }
+            else {
+                this.startDate = new TaskDate(startDate);
+                this.startTime = new TaskTime(startTime);
+                this.endDate = new TaskDate(endDate);
+                this.endTime = new TaskTime(endTime);
+                
+                if (!isValidDateInterval(this.startDate, this.endDate)) {
+                    throw new IllegalValueException(MESSAGE_INTERVAL_CONSTRAINTS_DATE);
+                }
+                
+                if (!isValidTimeInterval(this.startDate, this.startTime, this.endDate, this.endTime)) {
+                    throw new IllegalValueException(MESSAGE_INTERVAL_CONSTRAINTS_TIME);
+                }
+            } 
+        }
+    }
+
+    /**
+     * Returns true if a given interval has a valid task date interval.
```

Comment:
Use the `@return` statement for header comments for functions which return booleans. eg. `@return true if a given interval has a valid task date interval.`